File: src/domain/Document.ts
// ─── 文档领域类型 ───
// 纯数据定义，不含业务逻辑

export type ArticlePhase = "planning" | "writing" | "reviewing" | "complete";

export type SectionStatus = "pending" | "writing" | "complete" | "revised";

export interface OutlineSection {
  id: string;
  title: string;
  level: 1 | 2 | 3;
  description?: string;
  targetWordCount?: number;
  status: SectionStatus;
  notes?: string;
}

export interface ArticleStyleConfig {
  editorStyleTemplateId: string;
  lineHeight: number;
  editorFontSize: number;
  editorMaxWidth: number;
  editorParagraphGap: number;
  editorFontFamily: string;
  codeThemeId: string;
  macosCodeBlock: boolean;
  firstLineIndent: boolean;
  justifyAlign: boolean;
  headingConfig: Record<string, string[]>;
  bgPattern: string;
  accentColor: string;
  captionFormat: string;
  customCSS: string;
  articleThemeId: string;
}

export interface PublishRecord {
  id: string;
  articleId: string;
  platform: string;
  platformName?: string;
  platformArticleId?: string;
  status: "draft" | "published" | "failed";
  url?: string;
  publishTime?: number;
  publishedAt: number;
  platformUrl?: string;
  errorMessage?: string;
}

export interface ReviewSuggestion {
  sectionIndex: number;
  original: string;
  suggestion: string;
  accepted: boolean;
  rejected: boolean;
}

export interface ReviewState {
  reviewedAt: number;
  suggestions: ReviewSuggestion[];
  styleId: string;
  actionId: string;
}

export interface ArticleDocument {
  id: string;
  collectionId?: string;
  seriesId?: string;
  title: string;
  content: string;
  source?: "quick-start" | "ai-plan" | "series-plan";
  inspiration?: string;
  styleId: string;
  actionId: string;
  tone?: string;
  targetAudience?: string;
  targetWordCount?: number;
  phase: ArticlePhase;
  outline: OutlineSection[];
  tags: string[];
  styleConfig: ArticleStyleConfig;
  linkedFolder?: string;
  projectContext?: string;
  seriesContext?: string;
  publishRecords: PublishRecord[];
  reviewState?: ReviewState;
  reviewData?: Record<string, unknown>;
  version: number;
  createdAt: number;
  updatedAt: number;
}

/* ─── 默认值 ─── */

export const DEFAULT_STYLE_CONFIG: ArticleStyleConfig = {
  editorStyleTemplateId: "default",
  lineHeight: 1.75,
  editorFontSize: 15,
  editorMaxWidth: 820,
  editorParagraphGap: 1.25,
  editorFontFamily: "",
  codeThemeId: "atom-one-light",
  macosCodeBlock: false,
  firstLineIndent: false,
  justifyAlign: false,
  headingConfig: {},
  bgPattern: "",
  accentColor: "",
  captionFormat: "",
  customCSS: "",
  articleThemeId: "clean",
};

export function createDefaultDocument(
  id: string,
  title: string,
  overrides?: Partial<ArticleDocument>,
): ArticleDocument {
  const now = Date.now();
  return {
    id,
    title,
    content: "",
    styleId: "general",
    actionId: "action-write",
    phase: "planning",
    outline: [],
    tags: [],
    styleConfig: { ...DEFAULT_STYLE_CONFIG },
    publishRecords: [],
    version: 1,
    createdAt: now,
    updatedAt: now,
    ...overrides,
  };
}

File: src-tauri/src/domain/document.rs
// ─── Document 领域类型 ───

use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct OutlineSection {
    pub id: String,
    pub title: String,
    pub level: u32,
    pub description: Option<String>,
    pub target_word_count: Option<u32>,
    pub status: String,
    pub notes: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ArticleDocument {
    pub id: String,
    pub collection_id: Option<String>,
    pub series_id: Option<String>,
    pub title: String,
    pub content: String,
    pub style_id: String,
    pub action_id: String,
    pub tone: Option<String>,
    pub target_audience: Option<String>,
    pub target_word_count: Option<u32>,
    pub phase: String,
    pub outline: Vec<OutlineSection>,
    pub tags: Vec<String>,
    pub style_config: serde_json::Value,
    pub linked_folder: Option<String>,
    pub project_context: Option<String>,
    pub series_context: Option<String>,
    pub publish_records: Vec<serde_json::Value>,
    pub review_state: Option<serde_json::Value>,
    pub source: Option<String>,
    pub inspiration: Option<String>,
    pub version: u32,
    pub deleted_at: Option<u64>,
    pub created_at: u64,
    pub updated_at: u64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ArticleMeta {
    pub id: String,
    pub collection_id: String,
    pub title: String,
    pub created_at: u64,
    pub updated_at: u64,
    pub style_id: Option<String>,
    pub action_id: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct TrashItem {
    pub id: String,
    pub title: String,
    pub collection_id: String,
    pub collection_title: String,
    pub deleted_at: u64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ImageSavedResult {
    pub local_path: String,
    pub alt_text: String,
    pub revised_prompt: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ArticleBlueprint {
    pub working_title: String,
    pub description: String,
    pub target_word_count: Option<u32>,
    pub tone: Option<String>,
    pub target_audience: Option<String>,
    pub cover_image: Option<String>,
    pub phase: String,
    pub tags: Vec<String>,
    pub outline: Vec<OutlineSection>,
    pub updated_at: u64,
    pub style_id: Option<String>,
    pub action_id: Option<String>,
}
